### PR TITLE
Add support for the equivalent of the OpenSSL pkcs12 info option

### PR DESCRIPTION
**Pull request by timlegge**
# Description

This PR adds support for the equivalent of the OpenSSL pkcs12 info option.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Changes have been manually tested (please provide information on test platform using the fields below)

## Test / Development Platform Information

- Operating system and version Ubuntu 24.04 LTS
- Crypt::OpenSSL::PKCS12 version 1.10
- Perl version 5.38.2
- OpenSSL version 3.0.13

### CODE REVIEW

**Comment by timlegge** [comment]
@jonasbn I have updated the documentation and cleaned up the PR.  It is at the point where it needs a second set of eyes.  I have done my best to test on multiple openssl and architectures.  All the github actions pass and the latest release has been tested on HPUX and on OpenSSL 1.0.2k-fips, OpenSSL 1.1.1 w and OpenSSL 3.0.13 in particular.

It is a large PR and took a lot more time than I thought when I told @Tux that I would take a look so I know it will be an undertaking to review.  Here are some suggestions:

1. the info() and info_as_hash() are new
2. for the most part there were limited changes to existing functionality
- I had to remove the callback method and use extractBioString because threaded perls had problems and I was unable to resolve it.
- Most of the other changes were if/elsed based on whether INFO was being requested
3. Openssl code copied in had some modifications related to the threading support and passing HVs or other parameters that the code can update for the _as_hash functionality
4. I added a lot of tests - go figure, various p12 files have different contents and need separate tests
5. The original tests were only changed to remove the openssl version diag - I added a separate test for that

Ask away when you have time to review.  I am going to take  a break from this PR for now I any changes at this point.  I have to get back to some other code.

The only other thing I can think of is to add support to encrypt the private key that is output.  Currently it does the equivalent of -nodes,  This should be trivial to do with either a option for instantiating the PKCS12 or more likely as a part of the call to info() or info_as_hash().

I am also wondering if that should be info() for the hash and info_as_string() for the text output.

Tim

File: PKCS12.xs [hunk 1]
```diff
@@ -325,19 +519,437 @@ int dump_certs_keys_p12(BIO *bio, PKCS12 *p12, char *pass, int passlen, int opti
...
+  int i;
+  for (i = 0; i < len; i++) {
+    if (i == (len-1))
+      out += sprintf(out, "%02X", buf[i]);
+    else
+      out += sprintf(out, "%02X ", buf[i]);
+  }
+}
+
+void hex_prin(BIO *out, unsigned char *buf, int len)
```
Comment (jonasbn) [inline PKCS12.xs:639]: Hi @timlegge 

Was this method to be named: `hex_print` ?

**Review by jonasbn — CHANGES_REQUESTED** [review]
Hi @timlegge 

I have posted a single question, please address that and I think we are good to go.

File: PKCS12.xs (hunk 1 above)
Comment (timlegge) [reply PKCS12.xs:639]: That is a copy paste from openssl I believe.  https://github.com/openssl/openssl/blob/0cd9dd703ea575699b2d3cd74f1b8224447f4352/apps/pkcs12.c#L1320

**Comment by timlegge** [comment]
> Hi @timlegge
> 
> I have posted a single question, please address that and I think we are good to go.

@jonasbn I just replied.  It's a copy from https://github.com/openssl/openssl/blob/0cd9dd703ea575699b2d3cd74f1b8224447f4352/apps/pkcs12.c#L1320

**Review by jonasbn — APPROVED** [review]
This is approved

Thanks @timlegge

**Pull request merged**